crates/matrix-sdk/src/sliding_sync/view.rs

```diff
@@ -631,10 +631,17 @@ impl SlidingSyncViewRequestGenerator {
         limit: Option<u32>,
     ) -> v4::SyncRequestList {
         let calc_end = start + batch_size;
-        let end = match limit {
+
+        let mut end = match limit {
             Some(l) => std::cmp::min(l, calc_end),
             _ => calc_end,
         };
+
+        end = match self.view.rooms_count() {
+            Some(total_room_count) => end.clamp(0, std::cmp::max(0, total_room_count - 1)),
```

Comment:
This can be a simple  `cmp::min(end, total_room_count - 1)`. `clamp` with a lower bound of `0` and `max(0, _)` are pointless for unsigned integers.

Reply:
Right, gotcha 👍